geoportailv3/static/js/infobar/elevationdirective.js

```diff
@@ -46,34 +46,44 @@ app.module.directive('appElevation', app.elevationDirective);
 /**
  * @ngInject
  * @constructor
+ * @param {angular.Scope} $scope The scope.
  * @param {angular.$http} $http The angular http service.
  * @param {ngeo.Debounce} ngeoDebounce ngeoDebounce service.
  * @param {app.GetElevation} appGetElevation Elevation service.
+ * @param {ngeo.olcs.Service} ngeoOlcsService The service.
  */
 app.ElevationDirectiveController =
-    function($http, ngeoDebounce, appGetElevation) {
+    function($scope, $http, ngeoDebounce, appGetElevation, ngeoOlcsService) {
       var map = this['map'];
 
-  /**
-   * @type {app.GetElevation}
-   * @private
-   */
+      /**
+       * @type {app.GetElevation}
+       * @private
+       */
       this.getElevation_ = appGetElevation;
 
-  /**
-   * @type {string}
-   */
+      /**
+       * @type {string}
+       */
       this['elevation'] = '';
 
-      map.on('pointermove',
-      ngeoDebounce(
-      function(e) {
-        if (this['active']) {
-          this.getElevation_(e.coordinate).then(goog.bind(
-             function(elevation) {
-               this['elevation'] = elevation;
-             }, this
-             ));
+      let manager = ngeoOlcsService.getManager();
+
+      // 2D
+      map.on('pointermove', ngeoDebounce(function(e) {
+        if (!this['active']) {
+          return;
+        }
+        if (manager.is3dEnabled()) {
+          let coordinate = map.getCoordinateFromPixel(e.pixel);
+          if (!coordinate) {
+            return;
+          }
+          this['elevation'] = Math.round(coordinate[2]) + ' m';
+        } else {
+          this.getElevation_(e.coordinate).then(
```

Comment:
I think the luxembourg elevation is relative to the geoid. Which is not the case of Cesium.
So it is better to also retrieve the elevation through the service in all cases (to check if getElevation_ pushes or replace the 3d coordinate).